nexus/src/db/model.rs

```diff
@@ -101,6 +101,67 @@ macro_rules! impl_enum_type {
     }
 }
 
+macro_rules! impl_enum_type2 {
+    (
+        $(#[$enum_meta:meta])*
+        pub struct $diesel_type:ident;
+
+        $(#[$model_meta:meta])*
+        pub enum $model_type:ident;
+        $($enum_item:ident => $sql_value:literal)+
+    ) => {
+
+        $(#[$enum_meta])*
+        pub struct $diesel_type;
+
+        $(#[$model_meta])*
+        pub enum $model_type {
+            $(
+                $enum_item,
+            )*
+        }
+
+        impl<DB> ToSql<$diesel_type, DB> for $model_type
+        where
+            DB: Backend,
+        {
+            fn to_sql<W: std::io::Write>(
+                &self,
+                out: &mut serialize::Output<W, DB>,
+            ) -> serialize::Result {
+                match self {
+                    $(
+                    $model_type::$enum_item => {
+                        out.write_all($sql_value)?
+                    }
+                    )*
+                }
+                Ok(IsNull::No)
+            }
+        }
+
+        impl<DB> FromSql<$diesel_type, DB> for $model_type
+        where
+            DB: Backend + for<'a> BinaryRawValue<'a>,
+        {
+            fn from_sql(bytes: RawValue<DB>) -> deserialize::Result<Self> {
+                match DB::as_bytes(bytes) {
+                    $(
+                    $sql_value => {
+                        Ok($model_type::$enum_item)
```

Comment:
The original did this, i.e., `Ok(VpcRouterKind(external::VpcRouterKind::Custom))` as opposed to the simple `Ok(VpcRouterKind::Custom)`.

```rs
Ok($model_type(<$ext_type>::$enum_item))
```